packages/core/src/bind/connectFactoryObservable.ts

```diff
@@ -6,6 +6,52 @@ import { useObservable } from "../internal/useObservable"
 import { SUSPENSE } from "../SUSPENSE"
 import { takeUntilComplete } from "../internal/take-until-complete"
 
+class NestedMap<K extends [], V extends Object> {
```

Comment:
I'd move this in a separate file... I think it's a bit noisy here

Reply:
I decided to move it below (in the same file). The reason being that `NestedMap` is not generic enough to be consumed from anywhere else, it's very tightly coupled to this functionality. So, I decided to to move it below.